src/nsidc/metgen/lab/README.md

```diff
@@ -0,0 +1,90 @@
+# OLVIS1A Processor
+
+This script downloads OLVIS1A granules sequentially and generates premet and spatial files for them.
+
+## Usage
+
+```bash
+# Set authentication credentials (required for downloading data)
+# Option 1: Basic auth with username/password
+export EARTHDATA_USERNAME=your-username
+export EARTHDATA_PASSWORD=your-password
+
+# Option 2: Bearer token authentication (recommended)
+export EARTHDATA_TOKEN=your-bearer-token
+
+# Basic usage - process 5 granules
+metgenc-lab-olvis1a
+
+# Process a specific number of granules
+metgenc-lab-olvis1a -n 10
+
+# Specify output directory
+metgenc-lab-olvis1a -o my_output_dir
+
+# Combine options
+metgenc-lab-olvis1a -n 10 -o my_output
+```
+
+## Authentication
+
+The processor supports two authentication methods:
+
+1. **Bearer Token (Recommended)**
+   - Generate a token at https://urs.earthdata.nasa.gov/users/tokens
+   - Set environment variable: `export EARTHDATA_TOKEN="your-token-here"`
+
+2. **Basic Authentication**
+   - Set environment variables:
+     - `export EARTHDATA_USERNAME=your-username`
+     - `export EARTHDATA_PASSWORD=your-password`
+
+## Configuration
+
+The following values are hardcoded constants:
+- Environment: `prod` (https://cmr.earthdata.nasa.gov)
+- Collection: `OLVIS1A`
+- Version: `1`
+- Provider: `NSIDC_ECS`
+
+## Output Structure
+
+The script creates the following directory structure:
+
+```
+olvis1a_output/
+├── data/          # Downloaded granule data files (JPG images)
+├── premet/        # Generated premet files (.premet)
+└── spatial/       # Generated spatial files (.spatial)
+```
+
+## File Formats
+
+### Premet Files
+Contains metadata in key-value format:
+- Data filename
+- Version ID (001)
+- Begin/end dates and times
+- Collection short name (OLVIS1A)
+- Processing level (1A)
+- Platform/instrument information
```

Comment:
If there are additional attributes in the UMM-G, those would be handy to include in the `premet` output as well for a more accurate representation.

Reply:
Ah, I see you have hard-coded additional attribute values, so you can ignore my comment!